### PR TITLE
[hab3_merge] Updates in humanoid controller

### PR DESCRIPTION
**Pull request by xavierpuigf**
## Motivation and Context

Updates in the humanoid controller. Changes so that the humanoid can have relative offset transformations with a fixed base, needed for path planning with lower foot skating. Updated pytests to support humanoids with skin.

## How Has This Been Tested

Ran pytests.

## Types of changes

- **\[Bug Fix\]** (non-breaking change which fixes an issue)
- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [X] My code follows the code style of this project.
- [X] I have updated the documentation if required.
- [X] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [X] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-lab/habitat/articulated_agents/humanoids/kinematic_humanoid.py [hunk 1]
```diff
@@ -73,6 +75,26 @@ def __init__(
...
+            mn.Rad(self.offset_rot), mn.Vector3(0, 1.0, 0)
+        )
+        perm = mn.Matrix4.rotation(
+            mn.Rad(self.offset_rot), mn.Vector3(0, 0, 1.0)
+        )
+        self.offset_transform_base = perm @ add_rot
+
+        self.rest_joints = None
+
+    def set_rest_pose_path(self, rest_pose_path):
```
Comment (ASzot) [inline habitat-lab/habitat/articulated_agents/humanoids/kinematic_humanoid.py:89]: Why is this a publicly exposed method and the external agent manager is responsible for calling it? Can the KinematicHumanoid object just directly set the rest pose in its own constructor?

File: habitat-lab/habitat/tasks/rearrange/articulated_agent_manager.py [hunk 2]
```diff
@@ -77,6 +77,8 @@ def __init__(self, cfg, sim):
             agent_cfg = cfg.agents[agent_name]
             agent_cls = eval(agent_cfg.articulated_agent_type)
             agent = agent_cls(agent_cfg.articulated_agent_urdf, sim)
+            if agent_cfg.motion_data_path != "":
```
Comment (ASzot) [inline habitat-lab/habitat/tasks/rearrange/articulated_agent_manager.py:80]: I prefer not having this added complexity here and keeping the motion_data_path internal to the humanoid components.

**Review by ASzot — APPROVED** [review]
A minor comment, but other than that LGTM.

File: habitat-lab/habitat/articulated_agents/humanoids/kinematic_humanoid.py (hunk 1 above)
Comment (xavierpuigf) [reply habitat-lab/habitat/articulated_agents/humanoids/kinematic_humanoid.py:89]: Currently agents can only be passed the urdf_path. Is there a way we can pass more parameters? If so, I would pass the path to the resting pose to the constructor

**Pull request merged**